service/topology_coordinator.cc

```diff
@@ -2863,10 +2880,9 @@ future<bool> topology_coordinator::maybe_start_tablet_migration(group0_guard gua
 
     co_await generate_migration_updates(updates, guard, plan);
 
-    // We only want to consider transitioning into tablet resize finalization path, if there's no other work
-    // to be done (e.g. start migration or/and emit split decision).
-    if (updates.empty()) {
-        co_return co_await maybe_start_tablet_resize_finalization(std::move(guard), plan.resize_plan());
+    // Transition to tablet_migration iff migration work is pending or a tablet resize needs finalization.
+    if (updates.empty() && plan.resize_plan().finalize_resize.empty()) {
+        co_return false;
     }
```

Comment:
I think it's fine to keep the tablet resize finalization track for when no migrations are running, so topology metadata will be more meaningful, since no actual migrations are running. As far as I understand, Tomek wants to *also* allow finalization to be performed in the migration track (for anticipating resize finalization), but the resize transition state can still exist. @tgrabiec what do you think?

Reply:
done.